### ISSUE TITLE
Can you release detailed configuration?

### ISSUE THREAD

**Issue by csyanbin** [jakesnell/prototypical-networks#2]
Hi Jake,
    Prototypical networks is really a nice work.
    
    I have run this code to reproduce the results in NIPS paper. However, it seems the results have some differences with the paper.
    
    
NIPS2017 paper:

5way1shot|5way5shot|20way1shot|20way5shot
-- | -- | -- | --
98.8 | 99.7 | 96.0 | 98.9

Reproduced results:
-- | -- | -- | --
98.4 | 99.6 | 94.9 | 98.6

**Comment by bertinetto** [comment]
Hi, I got the same as @csyanbin with python 3.5, cuda 8 and pytorch 0.3

**Comment by csyanbin** [comment]
@bertinetto @jakesnell  
I also reproduced the experiments using tensorflow with the same setting and parameters, still got similar results. I wonder if I miss some implementation or parameter details?

**Comment by dnlcrl** [comment]
Hello, I just want to add that I implemented the same algorithm in a slight different way ([here](https://github.com/orobix/Prototypical-Networks-for-Few-shot-Learning-PyTorch)), and I got the same as @csyanbin too (except for the 20way1shot where I obtained 95.1%).

**Comment by csyanbin** [comment]
@dnlcrl I think the arxiv paper above cite the original prototypical paper of ICLR (https://openreview.net/references/pdf?id=BJ-3bnVmg), which is different setting with this git repo.
The ICLR version always uses 1-shot training, but this repo uses the corresponding shot.

**Comment by dnlcrl** [comment]
@csyanbin Oh I got it, thank you, I must have missed it.

**Comment by yannlif** [comment]
Hi,
@csyanbin I think the results reported are when training on both the training and validation set.
run_trainval.py run this full training using the epoch number validated on the validation set during the first training with the same other hyperparameters.

The results I get are closer to the reported ones but still different:

| 5way1shot | 5way5shot | 20way1shot | 20way5shot |
| -- | -- | -- | -- |
| 98.5 | 99.6 | 95.3 | 98.7 |

**Comment by csyanbin** [comment]
Hi,
@yannlif I think validation cannot be combined into the training set for fair comparisons. Also, in this paper, the authors said: "We follow their procedure by training on the 64 training classes and using the 16 validation classes for monitoring generalization performance only." 

Although the performance is slightly better, I think this is not fair for comparison.

**Comment by yannlif** [comment]
@csyanbin 

> We follow their procedure by training on the 64 training classes and using the 16 validation classes for monitoring generalization performance only.

This is for the mini-ImageNet experiments.
In the Omniglot part:

> We use 1200 characters plus rotations for training (4,800 classes in total) and the remaining classes, including rotations, for test.

This corresponds to the trainval split. The train split has only 1028 unique characters.

**Comment by csyanbin** [comment]
@yannlif 
I got it. I think you are right for Omniglot.
Thanks for this information. I will try again.

**Comment by debasmitdas** [comment]
Hi Guys,
Does anybody how many training epochs and episodes/epoch was used to reproduce the paper's results (for eg. the miniImagenet dataset) ?
